packages/database/src/migrations/20230222233428-RewriteVizsUsedDataElementCodeToName-modifies-schema.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const removeDataElementCodeToName = transform => {
+  const transformString = JSON.stringify(transform);
+  return JSON.parse(
+    transformString.replaceAll('dataElementCodeToName($dataElement)', '$dataElementName'),
+  );
+};
+
+const restoreDataElementCodeToName = transform => {
+  const transformString = JSON.stringify(transform);
+  return JSON.parse(
+    transformString.replaceAll('$dataElementName', 'dataElementCodeToName($dataElement)'),
+  );
+};
+
+exports.up = async function (db) {
+  const { rows: reports } = await db.runSql(
+    `SELECT * FROM report WHERE config::text LIKE '%dataElementCodeToName($dataElement)%'`,
+  );
+  for (const report of reports) {
+    const { transform } = report.config;
+    const { dataElementCodes } = transform[0].parameters;
+
+    const fetchMetaData = {
+      transform: 'fetchData',
+      dataTableCode: 'data_element_metadata',
+      parameters: { dataElementCodes },
+      join: [
+        {
+          tableColumn: 'dataElement',
+          newDataColumn: 'code',
+        },
+      ],
+    };
+    const excludeUnusedMetadata = {
+      transform: 'excludeColumns',
+      columns: ['id', 'code'],
+    };
+    const renameDataElementName = {
+      transform: 'updateColumns',
+      insert: {
+        dataElementName: '=$name',
+      },
+      exclude: ['name'],
+    };
+    const removeDataElementMetadata = {
+      transform: 'excludeColumns',
+      columns: ['dataElementName'],
+    };
+
+    let newTransform = transform
+      .map(t => {
+        if (JSON.stringify(t).includes('dataElementCodeToName(')) {
+          return [
+            fetchMetaData,
+            excludeUnusedMetadata,
+            renameDataElementName,
+            t,
+            removeDataElementMetadata,
```

Comment:
The idea is to add `fetchMetaData`, `excludeUnusedMetadata`, `renameDataElementName` in front of every transform that has `dataElementCodeToName`, then remove it afterward.

I know this is annoying, but if the metadata is added at the beginning of the transforms, it will crash most of the reports because it introduces a new column.